### PR TITLE
Created Room table for admin

### PR DESCRIPTION
**Pull request by 06hpathak**

### CODE REVIEW

File: src/hooks/use12CandleRoomDeletion.js
Comment (amitad16) [inline src/hooks/use12CandleRoomDeletion.js:1]: Please use `useMutation` for this

File: src/pages/admin/View12CandlesRoomsModal.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// ViewContentModal.js
+import React from 'react';
+import { Modal, Typography, Button } from '@mui/material';
+import RoomForm from './RoomForm';
+
+const ViewContentModal = ({ open, onClose, item }) => {
```
Comment (amitad16) [inline src/pages/admin/View12CandlesRoomsModal.js:6]: Rename this to `View12CandlesRoomModal`

File: src/pages/admin/Create12CandlesRoomModal.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { Modal, Button } from '@mui/material';
+import RoomForm from './RoomForm';
+import Typography from '@mui/material/Typography';
+
+const AddRoomModal = ({ open, onClose }) => {
```
Comment (amitad16) [inline src/pages/admin/Create12CandlesRoomModal.js:6]: Assign same naming to file name and component name

File: src/hooks/usePagination12CandlesRoom.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { useQuery } from '@tanstack/react-query';
+import TWELVE_CANDLES_GAME_ID from 'utils/constants';
```
Comment (amitad16) [inline src/hooks/usePagination12CandlesRoom.js:2]: Are you able to import it here?

But here you are not importing it.
https://github.com/thesiriusacademy/tsa-dashboard/pull/30/files#diff-b99195b45e9f1d84f4f48a688cba270d50b7782cfd24e6c310718cdad47b7298R3

**Review by amitad16 — CHANGES_REQUESTED** [review]
added comments

**Review by amitad16 — CHANGES_REQUESTED** [review]

File: src/pages/admin/View12CandlesRoomsModal.js (hunk 1 above)
Comment (06hpathak) [reply src/pages/admin/View12CandlesRoomsModal.js]: done

File: src/hooks/use12CandleRoomDeletion.js
Comment (06hpathak) [reply src/hooks/use12CandleRoomDeletion.js:1]: done

File: src/hooks/usePagination12CandlesRoom.js (hunk 3 above)
Comment (06hpathak) [reply src/hooks/usePagination12CandlesRoom.js:2]: done with different method

File: package.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
     "@testing-library/react": "^14.2.0",
     "@testing-library/user-event": "^14.5.2",
     "apexcharts": "^3.45.2",
+    "date-fns": "^3.6.0",
```
[the remark this answers is not in the document]
Comment (06hpathak) [reply package.json:23]: switched to moment

File: src/hooks/use12CandleRoomDeletion.js
[the remark this answers is not in the document]
Comment (06hpathak) [reply src/hooks/use12CandleRoomDeletion.js:1]: done

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (06hpathak) [reply package.json:23]: done

File: src/pages/admin/Create12CandlesRoomModal.js (hunk 2 above)
Comment (06hpathak) [reply src/pages/admin/Create12CandlesRoomModal.js:6]: done

File: src/hooks/useContentDeletion.js [hunk 5]
```diff
@@ -1,31 +1,23 @@
...
-          throw new Error('Failed to delete item');
-        }
-      } catch (error) {
-        console.error('Error deleting item:', error);
-      }
-    };
-
-    if (itemId) {
-      deleteContent();
+const delteContent = async ({ _id }) => {
```
Comment (amitad16) [inline src/hooks/useContentDeletion.js:3]: Name it `roomId` instead of `_id`

File: src/pages/admin/RoomForm.js [hunk 6]
```diff
@@ -5,24 +5,26 @@ import Alert from '@mui/material/Alert';
...
-  startDateTime: '',
-  endDateTime: '',
-  marketType: ''
-};
-
-const CreateRoomForm = () => {
+import useEdit12CandlesRooms from 'hooks/useEdit12CandlesRooms';
+const CreateRoomForm = ({ item, isViewable, isEditable }) => {
   const { mutateAsync } = useCreate12CandlesRoom();
+  const { mutateAsync: mutateAsync1 } = useEdit12CandlesRooms();
```
Comment (amitad16) [inline src/pages/admin/RoomForm.js:11]: Please rename the variables as below

```suggestion
  const { mutateAsync: mutateCreateRoomAsync } = useCreate12CandlesRoom();
  const { mutateAsync: mutateEditRoomAsync } = useEdit12CandlesRooms();
```

File: src/pages/admin/RoomForm.js [hunk 7]
```diff
@@ -65,21 +67,40 @@ const CreateRoomForm = () => {
...
     },
     onSubmit: async (values, { resetForm }) => {
-      try {
-        const data = await mutateAsync(values);
-        if (data) {
-          setSnackbarSeverity('success');
-          setSnackbarMessage('Room created successfully!');
+      if (isEditable) {
+        try {
+          const editdata = await mutateAsync1(values);
```
Comment (amitad16) [inline src/pages/admin/RoomForm.js:72]:
```suggestion
          const editdata = await mutateEditRoomAsync(values);
```

File: src/pages/admin/RoomForm.js [hunk 8]
```diff
@@ -65,21 +67,40 @@ const CreateRoomForm = () => {
...
+          }
+        } catch (err) {
+          console.error(err);
+          setSnackbarSeverity('error');
+          setSnackbarMessage('Failed to update room');
+          setOpenSnackbar(true);
+        }
+      } else {
+        try {
+          const data = await mutateAsync(values);
```
Comment (amitad16) [inline src/pages/admin/RoomForm.js:89]:
```suggestion
          const data = await mutateCreateRoomAsync(values);
```

**Review by amitad16 — CHANGES_REQUESTED** [review]
added some comments

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (amitad16) [reply package.json:23]: Please remove this package -> delete package-lock.json ->  and reinstall dependencies

File: src/pages/admin/View12CandlesRoomsModal.js (hunk 1 above)
Comment (amitad16) [reply src/pages/admin/View12CandlesRoomsModal.js]: I dont see it renamed....it is still `ViewContentModal`

Comment (06hpathak) [reply src/pages/admin/View12CandlesRoomsModal.js]: done

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (06hpathak) [reply package.json:23]: done

File: src/hooks/use12CandleRoomDeletion.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import { useMutation } from '@tanstack/react-query';
+import CONSTANTS from 'utils/constants';
+const delteRoom = async ({ _id }) => {
```
[the remark this answers is not in the document]
Comment (06hpathak) [reply src/hooks/use12CandleRoomDeletion.js:3]: done

File: src/hooks/useContentDeletion.js (hunk 5 above)
Comment (06hpathak) [reply src/hooks/useContentDeletion.js:3]: updated it as contentId

File: src/pages/admin/RoomForm.js (hunk 6 above)
Comment (06hpathak) [reply src/pages/admin/RoomForm.js:11]: done

File: src/pages/admin/RoomForm.js (hunk 7 above)
Comment (06hpathak) [reply src/pages/admin/RoomForm.js:72]: done

File: src/pages/admin/RoomForm.js (hunk 8 above)
Comment (06hpathak) [reply src/pages/admin/RoomForm.js:89]: done

**Review by amitad16 — APPROVED** [review]
LGTM

**Pull request merged**